### PR TITLE
[GCP Storage] Upgrade GCSFuse version

### PR DESCRIPTION
**Pull request by Michaelvll**
We encountered an issue with the gcsfuse that the saved large model checkpoints on a mounted GCS bucket can be incomplete. It could be a problem with the old gcsfuse we were using:
in the release note of gcsfuse (discovered by @infwinston):
> Fixed a critical bug in write workflow - although it was very rare, but gcsfuse was writing wrong file content while doing append operation on larger file.

https://github.com/GoogleCloudPlatform/gcsfuse/releases/tag/v0.41.11

Tested (run the relevant ones):

- [x] Any manual or new tests for this PR (please specify below)
  - [x] Launch a node and write things to the mounted bucket.
  - [x] `pytest tests/test_smoke.py::test_aws_storage_mounts --aws`; to test the mounting of GCS bucket on AWS VM.
  - [x] `pytest tests/test_smoke.py::test_gcp_storage_mounts`
- [x] All smoke tests: `pytest tests/test_smoke.py` 
- [ ] Relevant individual smoke tests: `pytest tests/test_smoke.py::test_fill_in_the_name` 
- [ ] Backward compatibility tests: `bash tests/backward_comaptibility_tests.sh`

### CODE REVIEW

**Comment by romilbhardwaj** [comment]
Thanks! Should we bump to `0.42.3` if we're upgrading versions?

**Comment by Michaelvll** [comment]
> Thanks! Should we bump to `0.42.3` (latest) if we're upgrading versions?

It is `0.42.3` in the PR changes?

File: sky/data/mounting_utils.py [hunk 1]
```diff
@@ -24,6 +26,9 @@ def get_mounting_command(
         str: Mounting command with the mounting script as a heredoc.
     """
     mount_binary = mount_cmd.split()[0]
+    installed_check = f'[ -x "$(command -v {mount_binary})" ]'
+    if version is not None:
+        installed_check += f' && {mount_binary} --version | grep -q {version}'
```
[the remark this answers is not in the document]
Comment (Michaelvll) [reply sky/data/mounting_utils.py]: Good point! Fixed. Thanks!

**Review by romilbhardwaj — APPROVED** [review]
Thanks for adding the version check! Should be good to go if `pytest tests/test_smoke.py::test_gcp_storage_mounts` passes.

**Pull request merged**